src/identity.js

```diff
@@ -0,0 +1,56 @@
+
+const isFunction = require('./utils/is-function')
+
+class Identity {
+  constructor (id, publicKey, provider) {
+    if (!id) {
+      throw new Error('Identity id is required')
+    }
+
+    if (!publicKey) {
+      throw new Error('Invalid public key')
+    }
+
+    if (!provider) {
+      throw new Error('Identity provider is required')
+    }
+
+    if (!provider.sign) {
+      throw new Error('Identity provider signing function is required')
+    }
+
+    if (!isFunction(provider.sign)) {
+      throw new Error('Identity provider signing function is invalid')
+    }
+
+    if (!provider.verify) {
+      throw new Error('Identity provider signature verification function is required')
+    }
+
+    if (!isFunction(provider.verify)) {
+      throw new Error('Identity provider signature verification function is invalid')
+    }
+
+    this._id = id
+    this._publicKey = publicKey
+    this._provider = provider
+  }
+
+  /**
+  * This is only used as a fallback to the clock id when necessary
+  * @return {string} public key hex encoded
+  */
+  get id () {
+    return this._id
+  }
+
+  get publicKey () {
+    return this._publicKey
+  }
+
+  get provider() {
+    return this._provider
+  }
+}
+
```

Comment:
It might make sense to add a convenience function here for signing, eg:
```javascript
sign (data) {
  return this.provider.sign(this, data)
}
```

The way I'm thinking it that the *Identity* can sign data (eg. "I prove that I created this data") and the *IdentityProvider* can verify that the data was signed by a certain Identity (as opposed to Identity verifying other identity's signature).

What do you think?